ckan/logic/action/update.py

```diff
@@ -123,6 +125,8 @@ def resource_update(context, data_dict):
     upload.upload(id, uploader.get_max_resource_size())
     model.repo.commit()
```

Comment:
if select_for_update we can expect some new errors from postgres here, right? We should catch those and turn them into validation errors we don't cause a server error and so the user knows to retry.